tests/monotouch-test/ARKit/ARPointCloudTest.cs

```diff
@@ -0,0 +1,93 @@
+﻿//
+// Unit tests for ARPointCloud
+//
+// Authors:
+//	Vincent Dondain <vidondai@microsoft.com>
+//
+// Copyright 2017 Microsoft. All rights reserved.
+//
+
+#if XAMCORE_2_0 && !__TVOS__ && !__WATCHOS__ && !MONOMAC
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using ARKit;
+using AVFoundation;
+using Foundation;
+using NUnit.Framework;
+using ObjCRuntime;
+using OpenTK;
+
+namespace MonoTouchFixtures.ARKit {
+
+	class ARPointCloudPoker : ARPointCloud {
+
+		public ARPointCloudPoker () : base (IntPtr.Zero)
+		{
+		}
+
+		public override nuint Count {
+			get {
+				return 2;
+			}
+		}
+
+		protected unsafe override IntPtr GetRawPoints ()
+		{
+			var va = new Vector3 [] { new Vector3 (1, 2, 3), new Vector3 (4, 5, 6) };
+			var gch = GCHandle.Alloc (va, GCHandleType.Pinned);
+			try {
+				Vector3* addr = (Vector3*)gch.AddrOfPinnedObject ();
+				return (IntPtr)addr;
+			} finally {
+				gch.Free ();
```

Comment:
This is not safe: you're freeing the data before returning a pointer to it to the caller.

You need to store the vector and the GCHandle in an instance variable, and free it in `Dispose`.